app/assets/javascripts/discourse/app/components/d-button.gjs

```diff
@@ -12,6 +17,54 @@ const ACTION_AS_STRING_DEPRECATION_ARGS = [
 ];
 
 export default class DButton extends GlimmerComponentWithDeprecatedParentView {
+  <template>
+    {{! template-lint-disable no-down-event-binding }}
+    <button
+      {{! For legacy compatibility. Prefer passing class as attributes. }}
+      class="{{@class}}
+        {{if @isLoading 'is-loading'}}
+        {{if this.btnLink 'btn-link' 'btn'}}
+        {{if this.noText 'no-text'}}
+        {{this.btnType}}"
+      {{! For legacy compatibility. Prefer passing these as html attributes. }}
+      id={{@id}}
+      form={{@form}}
+      aria-controls={{@ariaControls}}
+      aria-expanded={{this.computedAriaExpanded}}
+      tabindex={{@tabindex}}
+      type={{or @type "button"}}
+      ...attributes
+      disabled={{this.isDisabled}}
+      title={{this.computedTitle}}
+      aria-label={{this.computedAriaLabel}}
+      {{on "keydown" this.keyDown}}
+      {{on "click" this.click}}
+      {{on "mousedown" this.mouseDown}}
+    >
+      {{#if @isLoading}}
+        {{~icon "spinner" class="loading-icon"~}}
+      {{else}}
+        {{#if @icon}}
+          {{~icon @icon~}}
+        {{/if}}
+      {{/if}}
+
+      {{~#if this.computedLabel~}}
+        <span class="d-button-label">
+          {{~htmlSafe this.computedLabel~}}
+          {{~#if @ellipsis~}}
+            &hellip;
+          {{~/if~}}
+        </span>
+      {{~else if (not (has-block))~}}
+        &#8203;
+        {{! Zero-width space character, so icon-only button height = regular button height }}
+      {{~/if~}}
+
+      {{yield}}
```

Comment:
Personally, I'd probably refactor this into:

```hbs
      {{~#if this.computedLabel~}}
         <span class="d-button-label">
           {{~htmlSafe this.computedLabel~}}
           {{~#if @ellipsis~}}
             &hellip;
           {{~/if~}}
         </span>
       {{~else if (has-block)~}}
         {{yield}}
       {{~else~}}
         &#8203;
         {{! Zero-width space character, so icon-only button height = regular button height }}
       {{~/if~}}
```

(I don't think you can/should pass `@label`/`@translatedLabel` *and* a block at the same time? I actually didn't find any examples of a block being passed at all, but maybe it's used in plugins)

Since `computedLabel` is a getter anyway, the `htmlSafe` call could probably just go in there.

I think the first one is a good refactor for clarity/readability anyway, and the second one is a wash.

I didn't want to change the code unnecessarily for the purpose of this PR, but I'm brining these up to point out that there are often benign refactors that can help avoid some of the slightly rougher edges atm. In this particular case, those refactors would have avoided needing the `not` and `htmlSafe` helpers and the corresponding boilerplate changes to those files (but it wasn't a big deal to do it either). In general, with getters and plain function helpers being a pretty easy thing to do these days, I think the need for those template helpers are going to be less ubiquitous.